File: app/src/main/java/by/egorgutko/autorization/DI/DaggerController.java
package by.egorgutko.autorization.DI;

import by.egorgutko.autorization.DI.forAdd.ForAddTaskPresenterModule;
import by.egorgutko.autorization.DI.forAdd.ForAddTaskSubComponent;
import by.egorgutko.autorization.DI.forList.ForListPresenterModule;
import by.egorgutko.autorization.DI.forList.ForListTaskSubComponent;

public class DaggerController {
    private ForAddTaskSubComponent forAddTaskSubComponent;
    private ForListTaskSubComponent forListTaskSubComponent;
    private MyAppComponent myAppComponent;

    public DaggerController(MyAppComponent myAppComponent) {
        this.myAppComponent = myAppComponent;
    }

    public void initForAddTaskSubController() {
        if (forAddTaskSubComponent == null) {
            forAddTaskSubComponent = myAppComponent.add(new ForAddTaskPresenterModule());
        }
    }

    public ForListTaskSubComponent getForListTaskSubComponent() {
        return forListTaskSubComponent;
    }

    public ForAddTaskSubComponent getForAddTAskComponent() {
        return forAddTaskSubComponent;
    }

    public void initForListSubComponent() {
        if (forListTaskSubComponent == null) {
            forListTaskSubComponent = myAppComponent.add(new ForListPresenterModule());
        }
    }


}

File: app/src/main/java/by/egorgutko/autorization/DI/DaggerController.kt
package by.egorgutko.autorization.DI

import by.egorgutko.autorization.DI.forAdd.ForAddTaskPresenterModule
import by.egorgutko.autorization.DI.forAdd.ForAddTaskSubComponent
import by.egorgutko.autorization.DI.forList.ForListPresenterModule
import by.egorgutko.autorization.DI.forList.ForListTaskSubComponent

class DaggerController(private val myAppComponent: MyAppComponent) {
    var forAddTAskComponent: ForAddTaskSubComponent? = null
        private set
    var forListTaskSubComponent: ForListTaskSubComponent? = null
        private set

    fun initForAddTaskSubController() {
        if (forAddTAskComponent == null) {
            forAddTAskComponent = myAppComponent.add(ForAddTaskPresenterModule())
        }
    }

    fun initForListSubComponent() {
        if (forListTaskSubComponent == null) {
            forListTaskSubComponent = myAppComponent.add(ForListPresenterModule())
        }
    }


}
